### ISSUE TITLE
MB_SITE_URL env var does not work for site-url setting

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#9764]
As mentioned in #2090 

Settings defined with env vars are supposed to override any values in the cache/DB. However `site-url` has a custom magic getter function which calls `setting/get-string`, correctly; I suspect `setting/get-string` is just getting the cached/DB value instead of the env var value, which is not the behavior we want.

Investigate this issue and if my suspicions are correct fix `get-string` so Settings with custom magic getters will use env var values if they are present.

**Comment by tlrobinson** [comment]
Actually it seems like this *does* work:

```
docker run --rm -it -p3333:3000 -e MB_SITE_URL="http://metabase.localhost:3333/" metabase/metabase:v0.32.4
```

![Screenshot 2019-04-16 15 47 15](https://user-images.githubusercontent.com/18193/56248982-ed8ace00-605e-11e9-82fe-82038a25e5bd.png)

Maybe you need to include the trailing slash?

**Comment by camsaul** [comment]
Yeah, it should include the trailing slash. Right now it's the magic setter that adds the trailing slash if it's not there. So if you set via env var it won't add it and it won't work.

That's not predictable so I'll change that behavior

**Comment by camsaul** [comment]
Fixed by #9873
